### PR TITLE
chitter challenge

### PR DESCRIPTION
**Pull request by jrose111284**
sign in and out done

### CODE REVIEW

**Comment by jrose111284** [comment]
filtered some code deleted some non needed code

**Comment by jrose111284** [comment]
added read me plus all pages link together

**Comment by tansaku** [comment]
You had your chitter_challenge reviewed by **Pablo Domínguez**.
### The good points are:
* Correct installation instructions in README
* All unnecessary components removed, Ruby style guide conventions followed
* Rakefile has appropriate tasks
* Gemfile uses test groups
* spec_helper.rb pulls in entire app through single point, Only test helpers (not view helpers) are pulled in to spec_helper.rb
* Database Cleaner is set up correctly
* DataMapper set up in separate configuration file
* Feature tests organized correctly
* Feature and unit tests in appropriate locations
* Spec 'Helpers' used appropriately
* Asset routes set correctly
* flash.now and flash.next used correctly
* All controller methods short and succinct
* Forms use correct markup
* Correct use of Semantic HTML elements
* View Helpers used appropriately

### You should consider the following improvements:
* Prefer Modular over Classic Style
* Split Routes into Separate Controller files
* Use Partials with Standard HTML Conventions

### Additional comments:
Sign out link doesn't work


see https://github.com/makersacademy/chitter_challenge/blob/master/docs/review.md for more details

File: Rakefile [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+    puts "Auto-upgrade complete (no data loss)"
+  end
 
-task default: [:cop, :spec, :cuke]
\ No newline at end of file
+
+  desc "Destructive upgrade"
+  task :auto_migrate do
+    DataMapper.auto_migrate!
+    puts "Auto-migrate complete (data was lost)"
```
Comment (danldb) [inline Rakefile]: That could be a very scary message! I would maybe soften the wording a little bit (data may have been lost)

File: app/app.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+  enable :sessions
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+
+	get '/links' do
+		@links=Message.all
+		erb(:'links/index')
+	end
```
Comment (danldb) [inline app/app.rb]: indentation

File: app/app.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+require 'bcrypt'
+require 'sinatra/flash'
+
+class Chitter < Sinatra::Base
+  use Rack::MethodOverride
+  enable :sessions
+  set :session_secret, 'super secret'
+  register Sinatra::Flash
+
+	get '/links' do
```
Comment (danldb) [inline app/app.rb]: links?

File: app/app.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  post '/links' do
+  link = Message.new(message: params[:message])
+  link.save
+  redirect to('/links')
+end
+
+  get '/' do
+    redirect '/sign_up'
+  end
```
Comment (danldb) [inline app/app.rb]: what happens if user is already signed in?

File: app/app.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  post '/links/new' do
```
Comment (danldb) [inline app/app.rb]: this is not restful - should be a get request.

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def self.authenticate(email, password)
+  user = first(email: email)
+
+  if user && BCrypt::Password.new(user.password_digest) == password
+    user
+  else
+    nil
+  end
```
Comment (danldb) [inline app/models/user.rb]: no association between links and users

File: app/views/links/index.erb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<ul id='links'>
+	<% @links.each do |link| %>
+		<li>
+			<%= current_user.email %> Message: <%= link.message %>
+		</li>
+	<% end %>
+	<form action='/links/new' method='post'>
+	  <input type='submit' value='post a message'>
+	</form>
+</ul>
```
Comment (danldb) [inline app/views/links/index.erb]: indentation

File: app/views/links/index.erb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<ul id='links'>
+	<% @links.each do |link| %>
+		<li>
+			<%= current_user.email %> Message: <%= link.message %>
```
Comment (danldb) [inline app/views/links/index.erb]: So every message will look like it was posted by the current_user?

**Comment by danldb** [comment]
:construction: 

Thanks Jay - a few things to fix up, see inline comments.

**Pull request closed without merging**